src/Initializer/time_stepping/LtsWeights/LtsWeights.cpp

```diff
@@ -216,32 +235,38 @@ LtsWeights::ComputeWiggleFactorResult LtsWeights::computeBestWiggleFactor() {
 
   auto totalWiggleFactorReductions = 0u;
 
-  // Compute baseline cost before wiggle factor and merging of clusters
-  m_clusterIds = computeClusterIds(maxWiggleFactor);
-  if (ltsParameters->getWiggleFactorEnforceMaximumDifference()) {
-    totalWiggleFactorReductions += enforceMaximumDifference();
+  if (baselineCost) {
+    logInfo(rank) << "Baseline cost before cluster merging is" << *baselineCost;
+  } else {
+    // Compute baselineCost cost before wiggle factor and merging of clusters
+    totalWiggleFactorReductions +=
+        computeClusterIdsAndEnforceMaximumDifferenceCached(maxWiggleFactor);
+    baselineCost = computeGlobalCostOfClustering(m_clusterIds,
+                                                 m_cellCosts,
+                                                 m_rate,
+                                                 maxWiggleFactor,
+                                                 m_details.globalMinTimeStep,
+                                                 MPI::mpi.comm());
+    logInfo(rank) << "Baseline cost, without wiggle factor and cluster merging is" << *baselineCost;
   }
-  const double baselineCost = computeGlobalCostOfClustering(
-      m_clusterIds, m_cellCosts, m_rate, maxWiggleFactor, m_details.globalMinTimeStep, MPI::mpi.comm());
+  assert(baselineCost);
+
   const double maxAdmissibleCost =
```

Comment:
I would move this line in 
```
 if (isAutoMergeUsed) {
 ```

Reply:
IMO it's better after the if statement as this checks that we have a non-nothing baseline cost.